### PR TITLE
avro-cpp: 1.8.2 -> 1.10.2

### PR DESCRIPTION
**Pull request by nagy**
###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129416` at 788a4c2c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/498fd7a2dc3cf23ff6a0e735d4011301">irods</a></li>
    <li><a href="https://gist.github.com/8867dfd20ba0ae16dce23454ba443cff">irods-icommands</a></li>
    <li><a href="https://gist.github.com/d95be53a699e988f416e0197323b278f">kafkacat</a></li>
    <li><a href="https://gist.github.com/b4ed72b44287cee5148fa524bcf75e99">libserdes</a></li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>avro-cpp</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by SuperSandro2000** [comment]
This bump breaks libserdes.

**Comment by nagy** [comment]
> This bump breaks libserdes.

Maybe we can apply https://github.com/confluentinc/libserdes/pull/17 as a patch to `libserdes`.

**Comment by SuperSandro2000** [comment]
> Maybe we can apply [confluentinc/libserdes#17](https://github.com/confluentinc/libserdes/pull/17) as a patch to `libserdes`.

Can you try that in this PR?

**Comment by nagy** [comment]
Done. Seems to have worked.

File: pkgs/development/libraries/libserdes/default.nix [hunk 1]
```diff
@@ -28,6 +29,14 @@ stdenv.mkDerivation rec {
 
   makeFlags = [ "GEN_PKG_CONFIG=y" ];
 
+  patches = [
+    # Fix compatibility with Avro master branch
+    (fetchpatch {
+      url = "https://github.com/confluentinc/libserdes/pull/17.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libserdes/default.nix:35]: The hash of that URL could change in the future. Using the follow is advised. 

```suggestion
      url = "https://github.com/confluentinc/libserdes/commit/d7a355e712ab63ec77f6722fb5a9e8056e7416a2.patch";
```

Comment (nagy) [reply pkgs/development/libraries/libserdes/default.nix]: Updated. But i guess this only works because there is only one commit.

**Review by liff — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@nagy Can you take a look at the irods build failure, too?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 129416` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>irods: log was empty</li>
<li>irods-icommands: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>avro-cpp</li>
<li>kafkacat</li>
<li>libserdes</li>
</ul>
</details>

**Comment by nagy** [comment]
> Can you take a look at the irods build failure, too?

I tried fixing that and it seems like we need to bump it to 4.2.9 . However,
that requires `nanodbc`, which is not packaged yet. I can make an issue for
that.

https://github.com/nanodbc/nanodbc

**Comment by risicle** [comment]
`nixpkgs-review` passes, macos 10.15. Same `irods` failure as everyone else on linux, should perhaps hold off until that's been dealt with?

**Comment by c0bw3b** [comment]
Friendly ping to @bzizou > any chance that `irods` package gets updated? :)

**Comment by bzizou** [comment]
@c0bw3b Oh! I'm going to check that asap!

**Comment by liff** [comment]
The direct dependencies build fine:

```
3 packages built:
avro-cpp kcat libserdes
```

The `irods` package doesn’t build on master (on Linux?) at all, with or without this update:

```
at pkgs/development/libraries/boost/generic.nix:35:1:

           34| # Boost <1.69 can't be built on linux with clang >8, because pth was removed
           35| assert with lib; ((stdenv.isLinux && toolset == "clang" && !(versionOlder stdenv.cc.version "8.0.0")) -> !(versionOlder version "1.69"));
             | ^
           36|
```

**Comment by bzizou** [comment]
@liff yes, I'm on it...

**Comment by bzizou** [comment]
So, I have locally an updated irods branch with irods 4.2.11 that builds correctly, but that depends on #161196 and on #129416 (this PR)...

**Comment by bzizou** [comment]
#161196 should be fixed for darwin... I don't have much skills for that... someone else to help?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 129416` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>avro-cpp</li>
<li>kcat</li>
<li>libserdes</li>
</ul>
</details>

**Pull request merged**